system/loggerd/encoderd.cc

```diff
@@ -127,12 +124,27 @@ void encoder_thread(EncoderdState *s, const LogCameraInfo &cam_info) {
 void encoderd_thread() {
   EncoderdState s;
 
-  std::vector<std::thread> encoder_threads;
-  for (const auto &cam : cameras_logged) {
-    encoder_threads.push_back(std::thread(encoder_thread, &s, cam));
-    s.max_waiting++;
+  std::set<VisionStreamType> streams;
+  while (!do_exit) {
```

Comment:
we have this loop in so many places now, would be nice to make the blocking version handle this